gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_util.py

```diff
@@ -0,0 +1,88 @@
+# !/usr/bin/env python3
+###############################################################################
+# $Id$
+#
+#  Project:  GDAL utils.auxiliary
+#  Purpose:  OSR utility functions
+#  Author:   Idan Miara <idan@miara.com>
+#
+# ******************************************************************************
+#  Copyright (c) 2021, Idan Miara <idan@miara.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+import numpy as np
+from typing import Union, Tuple, Optional
+
+import osgeo
+from osgeo import osr, ogr, gdal
+
+AnySRS = Union[str, int, osr.SpatialReference, gdal.Dataset]
+
+
+def get_srs(srs: AnySRS, gis_order: bool = True) -> osr.SpatialReference:
+    """ returns an SRS object from epsg, pj_string or DataSet or SRS object """
+    if isinstance(srs, osr.SpatialReference):
+        pass
+    elif isinstance(srs, gdal.Dataset):
+        srs = get_srs_from_ds(srs)
+    elif isinstance(srs, int):
+        srs_ = osr.SpatialReference()
+        if srs_.ImportFromEPSG(srs) != ogr.OGRERR_NONE:
+            raise Exception(f'ogr error when parsing srs from epsg:{srs}')
+        srs = srs_
+    elif isinstance(srs, str):
+        srs_ = osr.SpatialReference()
+        if srs_.ImportFromProj4(srs) != ogr.OGRERR_NONE:
+            raise Exception(f'ogr error when parsing srs from proj string: {srs}')
+        srs = srs_
+    else:
+        raise Exception(f'Unknown SRS: {srs}')
+    if gis_order and int(osgeo.__version__[0]) >= 3:
+        # GDAL 3 changes axis order: https://github.com/OSGeo/gdal/issues/1546
+        srs.SetAxisMappingStrategy(osgeo.osr.OAMS_TRADITIONAL_GIS_ORDER)
```

Comment:
I would not do forcing traditional GIS order in a utility function. Upper layers can do that if they wish, but I'd expect utility classes to be axis order compliant by default.  And we don't need to do the version checking as nobody < GDAL 3.3 is supposed to use that

Reply:
Would it be OK to leave an optional gis_order variable with no default or with False default?
I think that the cpp version of `gdallocationinfo` -wgs84 uses GIS order, and I wanted the py version to be compatible.

Although it's not advertised, I believe most of `gdal_utils` is compatible with much lower versions of gdal, we could try to keep compatibility if it's not complicated, as in this case.

Reply:
> Would it be OK to leave an optional gis_order variable with no default or with False default?

ok